ids/format_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package ids
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestFormat(t *testing.T) {
+	type (
+		idInterface interface {
+			String() string
+			Hex() string
+		}
+		test struct {
+			id   idInterface
+			want map[string]string // format -> output
+		}
+	)
+	makeTestCase := func(id idInterface) test {
+		return test{
+			id: id,
+			want: map[string]string{
+				"%v":  id.String(),
+				"%s":  id.String(),
+				"%q":  `"` + id.String() + `"`,
+				"%x":  id.Hex(),
+				"%#x": `0x` + id.Hex(),
+			},
+		}
+	}
+
+	tests := []test{
+		makeTestCase(ID{}),
+		makeTestCase(GenerateTestID()),
+		makeTestCase(GenerateTestID()),
+		makeTestCase(ShortID{}),
+		makeTestCase(GenerateTestShortID()),
+		makeTestCase(GenerateTestShortID()),
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.id.String(), func(t *testing.T) {
+			for format, want := range tt.want {
+				if got := fmt.Sprintf(format, tt.id); got != want {
+					//nolint:forbidigo // require.Equal() is inappropriate as it unnecessarily stops future tests and `assert.Equal()` isn't allowed
+					t.Errorf("fmt.Sprintf(%q, %T) got %q; want %q", format, tt.id, got, want)
+				}
```

Comment:
```suggestion
				t.Run(format, func(t *testing.T) {
					got := fmt.Sprintf(format, tt.id)
					require.Equal(t, want, got)
				})
```

Reply:
`require` / `t.Fatal` stop all further tests, which IMO is bad practice when used as a blanket rule because it hides possibly useful information.

For example (granted, a very trivial one) did my test fail because I convert to hex incorrectly, or because I forgot to add the `0x` prefix? Seeing multiple failures gives the full context instead of forcing the engineer to play whack-a-mole.

WDYT? Is this a battle that's already been fought and won/lost in this code base?

Reply:
> Is this a battle that's already been fought and won/lost in this code base?

Yes

Reply:
![image](https://github.com/user-attachments/assets/cc4fd97c-ca91-4b4a-bcc1-b24501d71123)